File: lecture_003(Pattern)/l_005_pattern5.java
import java.util.Scanner;
public class l_005_pattern5 {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        patternDiamondNumber(n);
    }
    public static void patternDiamondNumber(int n) {
        int nst=1;
        int nsp=n/2;
        for (int r=1; r<=n; r++) {
            for (int csp=1; csp<=nsp; csp++) {
                System.out.print(" ");
            }
            int val = r;
            for (int cst=1; cst<=nst; cst++) {
                System.out.print(val);
            }
            if(r <= n/2) {
                nst+=2;
                nsp--;
            }
            else {
                nst-=2;
                nsp++;
            }
            System.out.println();
        }
    }
}

File: lecture_003(Pattern)/l_005_pattern5.cpp
#include<iostream>
using namespace std;

void patterndiamondnumber(int n){
    int nst=1;
    int nsp=n/2;
    int num=1; 
    for (int r=1; r<=n; r++) {
        for (int csp=1; csp<=nsp; csp++) {
           cout<<" ";
        }
        for (int cst=1; cst<=nst; cst++) {
            cout<<num;
        }
        if(r<=n/2) {
            nst+=2;
            nsp--;
        }
        else {
            nst-=2;
            nsp++;
        }
        num++;
        cout<<endl;
    }
}

int main(int args, char**argv) {
    int n;
    cin>>n;
    patterndiamondnumber(n);
    return 0;
}